File: examples/Python/mmiecho.py
"""
MMI echo query example

Author : Min RK <benjaminrk@gmail.com>

"""

import sys
from mdcliapi import MajorDomoClient

def main():
    verbose = '-v' in sys.argv
    client = MajorDomoClient("tcp://localhost:5555", verbose)
    request = "echo"
    reply = client.send("mmi.service", request)
    
    if reply:
        replycode = reply[0]
        print "Lookup echo service:", replycode
    else:
        print "E: no response from broker, make sure it's running"
    
if __name__ == '__main__':
    main()

File: examples/Go/mmiecho.go
//
// MMI echo query example
//
// Author: amyangfei <amyangfei@gmail.com>
// Requires: http://github.com/alecthomas/gozmq

package main

import (
    "fmt"
    "os"
)

func main() {
    verbose := len(os.Args) > 1 && os.Args[1] == "-v"
    client := NewClient("tcp://localhost:5555", verbose)
    defer client.Close()

    // This is the service we send our request to
    reply := client.Send([]byte("mmi.service"), [][]byte{[]byte("echo")})

    fmt.Println("here")
    if len(reply) > 0 {
        reply_code := reply[0]
        fmt.Printf("Lookup echo service: %s\n", reply_code)
    } else {
        fmt.Printf("E: no response from broker, make sure it's running\n")
    }
}
